trace-reader/src/main.rs

```diff
@@ -0,0 +1,85 @@
+use anyhow::Result;
+use clap::Parser;
+use rand::{seq::IteratorRandom, thread_rng};
+use rdkafka::producer::FutureProducer;
+use std::net::SocketAddr;
+use trace_reader::{dispatch_trace_file, load_trace_file};
+
+// cargo run -- --broker localhost:19092
+
+#[derive(Debug, Parser)]
+#[clap(author, version, about)]
+struct Cli {
+    #[clap(short, long)]
+    broker: String,
+
+    #[clap(short, long)]
+    username: Option<String>,
+
+    #[clap(short, long)]
+    password: Option<String>,
+
+    #[clap(short, long = "group", default_value = "trace-producer")]
+    consumer_group: String,
+
+    #[clap(short, long, default_value = "Traces")]
+    trace_topic: String,
+
+    #[clap(short, long, default_value = "127.0.0.1:9090")]
+    observability_address: SocketAddr,
+
+    #[clap(short, long)]
+    file_name: Option<String>,
+
+    #[clap(short, long, default_value = "1")]
+    number_of_events: usize,
+
+    #[clap(short, long, default_value = "false")]
+    random_sample: bool,
+}
+
+#[tokio::main]
+async fn main() -> Result<()> {
+    env_logger::Builder::from_env(env_logger::Env::default().default_filter_or("info")).init();
+
+    let args = Cli::parse();
+
+    let file_name = args.file_name.unwrap_or(
```

Comment:
This is overly specific for a default value, we should just require a file to be specified.